pkg/kvstore/etcd.go

```diff
@@ -1270,7 +1322,12 @@ func (e *etcdClient) GetIfLocked(ctx context.Context, key string, lock KVLocker)
 	defer func() { Trace("GetIfLocked", err, logrus.Fields{fieldKey: key, fieldValue: string(bv)}) }()
 
 	duration := spanstat.Start()
-	e.limiter.Wait(ctx)
+	lr, err := e.limiter.Wait(ctx)
+	if err != nil {
+		increaseMetric(key, metricRead, "GetLocked", duration.EndError(err).Total(), err)
```

Comment:
We cannot with the current `increaseMetric` implementation, since `duration.EndError(err).Total()` would be executed immediately, hence not counting correctly the duration. It should work modifying `increaseMetric` to take the span as input (rather than the pre-computed duration), but the change tends to be on the large side. WDYT?

Reply:
ah yea, we could do that in separate PR.

Reply:
I'm giving it a try. If it ends up being too large I'll open a separate PR

Reply:
I've updated the PR adding one more commit which does the `increaseMetrics` deferring, to avoid having to repeat that in case of errors. @marseel PTAL